Sources/TuistPlugin/PluginService.swift

```diff
@@ -0,0 +1,89 @@
+import TSCBasic
+import TuistCore
+import TuistLoader
+import TuistSupport
+
+/// A default implementation of `PluginServicing` which loads `Plugins` using
+/// the `Config` description without it needing to be loaded. The  plugins are first fetched/copied into
+/// a cache and the loaded `Plugins` model is returned.
+public final class PluginService: PluginServicing {
+    private let modelLoader: GeneratorModelLoading
+    private let fileHandler: FileHandling
+    private let gitHandler: GitHandling
+    private let manifestFilesLocator: ManifestFilesLocating
+
+    /// Creates a `PluginService`.
+    /// - Parameters:
+    ///   - modelLoader: A model loader for loading plugins.
+    ///   - fileHandler: A file handler for creating plugin directories/related files.
+    ///   - gitHandler: A git handler for cloning and interacting with remote plugins.
+    ///   - manifestFilesLocator: A locator for manifest files, used to find location of `Config` manifest in order to load plugins.
+    public init(
+        modelLoader: GeneratorModelLoading = GeneratorModelLoader(manifestLoader: ManifestLoader(), manifestLinter: ManifestLinter()),
+        fileHandler: FileHandling = FileHandler.shared,
+        gitHandler: GitHandling = GitHandler(),
+        manifestFilesLocator: ManifestFilesLocating = ManifestFilesLocator()
+    ) {
+        self.modelLoader = modelLoader
+        self.fileHandler = fileHandler
+        self.gitHandler = gitHandler
+        self.manifestFilesLocator = manifestFilesLocator
+    }
+
+    public func loadPlugins(at path: AbsolutePath) throws -> Plugins {
+        guard let configPath = manifestFilesLocator.locateConfig(at: path) else {
+            throw PluginServiceError.configNotFound(path)
+        }
+
+        let config = try modelLoader.loadConfig(at: configPath)
+        return try loadPlugins(using: config)
+    }
+
+    public func loadPlugins(using config: Config) throws -> Plugins {
+        let pluginPaths = try fetchPlugins(config: config)
+        let pluginManifests = try pluginPaths.map(modelLoader.loadPlugin)
+        let projectDescriptionHelpers = zip(pluginManifests, pluginPaths)
+            .compactMap { plugin, path -> ProjectDescriptionHelpersPlugin? in
+                let helpersPath = path.appending(RelativePath(Constants.helpersDirectoryName))
+                guard fileHandler.exists(helpersPath) else { return nil }
+                return ProjectDescriptionHelpersPlugin(name: plugin.name, path: helpersPath)
+            }
+
+        return Plugins(projectDescriptionHelpers: projectDescriptionHelpers)
+    }
+
+    private func fetchPlugins(config: Config) throws -> [AbsolutePath] {
+        try config.plugins
+            .map { plugin in
+                switch plugin {
+                case let .local(path):
+                    logger.debug("Fetching \(plugin.description) at: \(path)")
+                    return AbsolutePath(path)
+                case let .gitWithBranch(url, id),
```

Comment:
Support for branches requires further work because we might have a local copy pointing to a commit X, but in remote there's a more recent commit Y that we might want to fetch. For this first implementation, I'd only support tags and shas. Once we see how users use the feature, if they ask for branches support, we can invest some efforts into enabling it. What do you think?